libraries/chain/controller.cpp

```diff
@@ -4606,14 +4606,23 @@ struct controller_impl {
       return applied_trxs;
    }
 
-   void interrupt_transaction() {
+   void interrupt_transaction(controller::interrupt_t interrupt) {
       // Do not interrupt during replay. ctrl-c during replay is handled at block boundaries.
       // Interrupt both speculative trxs and trxs while applying a block.
       // This is to allow killing a long-running transaction in a block being validated during apply block.
       // This also allows killing a trx when a block is received to prioritize block validation.
       if (!replaying) {
-         dlog("Interrupting trx...");
-         main_thread_timer.interrupt_timer();
+         if (applying_block) {
+            if (interrupt == controller::interrupt_t::all_trx || interrupt == controller::interrupt_t::apply_block_trx) {
```

Comment:
Is `interrupt` set to `interrupt_t::apply_block_trx` only in tests (except `all` implies it)?

Reply:
Yes. Don't currently have a use case of interrupt `apply_block_trx` that is not all.